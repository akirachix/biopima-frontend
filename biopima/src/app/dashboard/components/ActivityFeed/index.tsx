import { Thermometer, AlertTriangle } from 'lucide-react';

export default function ActivityFeed() {
  return (
    <div className="bg-white rounded-3xl p-4 sm:p-6 shadow-lg border border-green-300 max-w-sm">
      <h3 className="text-lg font-bold text-green-900 mb-1">Activity Feed</h3>
      <p className="text-xs text-green-700 mb-4">Recent system events and alerts.</p>

      <div className="space-y-5">
        <div className="flex flex-col">
          <div className="flex items-center mb-1 space-x-2">
            <span className="w-3 h-3 rounded-full bg-green-500"></span>
            <div className="flex items-center space-x-1 text-green-700">
              <Thermometer className="w-5 h-5 flex-shrink-0" />
              <strong className="text-base select-none">Temp Warning</strong>
            </div>
            <span className="ml-auto text-xs text-gray-500 select-none">
              less than a minute ago
            </span>
          </div>
          <p className="ml-5 text-sm text-gray-700 leading-snug">
            Low Temperature Warning: Digester temperature at 34.6Â°C.
          </p>
        </div>

        <div className="flex flex-col">
          <div className="flex items-center mb-1 space-x-2">
            <span className="w-3 h-3 rounded-full bg-yellow-700"></span>
            <div className="flex items-center space-x-1 text-yellow-700">
              <AlertTriangle className="w-5 h-5 flex-shrink-0" />
              <strong className="text-base select-none">Leak Warning</strong>
            </div>
            <span className="ml-auto text-xs text-gray-500 select-none">
              less than a minute ago
            </span>
          </div>
          <p className="ml-5 text-sm text-gray-700 leading-snug">
            Leak Warning: Pressure at 1.02 bar (15.0% drop).
          </p>
        </div>
      </div>
    </div>
  );
}
