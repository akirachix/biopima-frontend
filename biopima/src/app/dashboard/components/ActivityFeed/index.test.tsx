
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ActivityFeed from '.';

describe('ActivityFeed', () => {
  it('renders title and description', () => {
    render(<ActivityFeed />);

    const title = screen.getByText(/Activity Feed/i);
    expect(title).toBeInTheDocument();
    expect(title).toHaveClass('text-green-900');

    const description = screen.getByText(/Recent system events and alerts./i);
    expect(description).toBeInTheDocument();
    expect(description).toHaveClass('text-green-700');
  });

  it('renders temperature warning activity item', () => {
    render(<ActivityFeed />);

    const tempDot = screen.getAllByRole('presentation')[0]; 
    expect(tempDot).toHaveClass('bg-green-500');

    const thermometerIcon = screen.getByLabelText(/thermometer/i);
    expect(thermometerIcon).toBeInTheDocument();

    const tempTitle = screen.getByText(/Temp Warning/i);
    expect(tempTitle).toBeInTheDocument();
    expect(tempTitle).toHaveClass('text-green-700');

    const tempTime = screen.getByText(/less than a minute ago/i);
    expect(tempTime).toBeInTheDocument();

    const tempDesc = screen.getByText(/Low Temperature Warning: Digester temperature at 34.6Â°C./i);
    expect(tempDesc).toBeInTheDocument();
    expect(tempDesc).toHaveClass('text-gray-700');
  });

  it('renders leak warning activity item', () => {
    render(<ActivityFeed />);

    const leakDot = screen.getAllByRole('presentation')[1];
    expect(leakDot).toHaveClass('bg-yellow-700');

    const alertIcon = screen.getByLabelText(/alert triangle/i);
    expect(alertIcon).toBeInTheDocument();

    const leakTitle = screen.getByText(/Leak Warning/i);
    expect(leakTitle).toBeInTheDocument();
    expect(leakTitle).toHaveClass('text-yellow-700');

    const leakTime = screen.getByText(/less than a minute ago/i);
    expect(leakTime).toBeInTheDocument();

    const leakDesc = screen.getByText(/Leak Warning: Pressure at 1.02 bar \(15.0% drop\)./i);
    expect(leakDesc).toBeInTheDocument();
    expect(leakDesc).toHaveClass('text-gray-700');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<ActivityFeed />);
    expect(asFragment()).toMatchSnapshot();
  });
});