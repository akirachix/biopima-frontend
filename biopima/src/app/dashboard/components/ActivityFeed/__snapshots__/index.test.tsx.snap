// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ActivityFeed matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white rounded-3xl p-4 sm:p-6 shadow-lg border border-green-300 max-w-sm"
  >
    <h3
      class="text-lg font-bold text-green-900 mb-1"
    >
      Activity Feed
    </h3>
    <p
      class="text-xs text-green-700 mb-4"
    >
      Recent system events and alerts.
    </p>
    <div
      class="space-y-5"
    >
      <div
        class="flex flex-col"
      >
        <div
          class="flex items-center mb-1 space-x-2"
        >
          <span
            class="w-3 h-3 rounded-full bg-green-500"
          />
          <div
            class="flex items-center space-x-1 text-green-700"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-thermometer w-5 h-5 flex-shrink-0"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 4v10.54a4 4 0 1 1-4 0V4a2 2 0 0 1 4 0Z"
              />
            </svg>
            <strong
              class="text-base select-none"
            >
              Temp Warning
            </strong>
          </div>
          <span
            class="ml-auto text-xs text-gray-500 select-none"
          >
            less than a minute ago
          </span>
        </div>
        <p
          class="ml-5 text-sm text-gray-700 leading-snug"
        >
          Low Temperature Warning: Digester temperature at 34.6Â°C.
        </p>
      </div>
      <div
        class="flex flex-col"
      >
        <div
          class="flex items-center mb-1 space-x-2"
        >
          <span
            class="w-3 h-3 rounded-full bg-yellow-700"
          />
          <div
            class="flex items-center space-x-1 text-yellow-700"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-triangle-alert w-5 h-5 flex-shrink-0"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m21.73 18-8-14a2 2 0 0 0-3.48 0l-8 14A2 2 0 0 0 4 21h16a2 2 0 0 0 1.73-3"
              />
              <path
                d="M12 9v4"
              />
              <path
                d="M12 17h.01"
              />
            </svg>
            <strong
              class="text-base select-none"
            >
              Leak Warning
            </strong>
          </div>
          <span
            class="ml-auto text-xs text-gray-500 select-none"
          >
            less than a minute ago
          </span>
        </div>
        <p
          class="ml-5 text-sm text-gray-700 leading-snug"
        >
          Leak Warning: Pressure at 1.02 bar (15.0% drop).
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
